stdlib/public/core/Array.swift

```diff
@@ -1865,3 +1869,189 @@ internal struct _ArrayAnyHashableBox<Element: Hashable>
     return true
   }
 }
+
+// SWIFT_ENABLE_TENSORFLOW
+extension Array where Element : Differentiable {
+  /// Views the array as the differentiable product manifold of `Element` multiplied with itself
+  /// `count` times.
+  @_fixed_layout
+  public struct DifferentiableView : Differentiable {
+    private var base: [Element]
+
+    /// The viewed array.
+    // I'm implementing this as a computed property instead of directly exposing `base` because the
+    // `@differentiable` annotation does not make the stored property actually differentiable. I
+    // think this is a bug. Maybe it's related to `@_fixed_layout`?
+    // TODO: Determine if that is a bug, and fix.
```

Comment:
Interesting. How about renaming the current `base` to `_base`, and renaming `array`/`_vjpArray` to `base`/`_vjpBase`?